gossip/evmstore/store_test.go

```diff
@@ -15,3 +20,14 @@ func nonCachedStore() *Store {
 
 	return NewStore(memorydb.NewProducer(""), cfg)
 }
+
+func TestStoreSetTx(t *testing.T) {
+	store := cachedStore()
+
+	tx := types.NewTx(&types.LegacyTx{Data: []byte("test")})
+
+	store.SetTx(tx.Hash(), tx)
+
+	txFromStore := store.GetTx(tx.Hash())
+	assert.Equal(t, tx.Data(), txFromStore.Data())
```

Comment:
How about comparing of the rest of tx's public fields?

Reply:
Good call, added